### PR TITLE
Update gitignore for IntelliJ module files

### PR DESCRIPTION
**Pull request by JonasKunz**
Title says it all.

### CODE REVIEW

**Comment by dmathieu** [comment]
If other repositories are adding this, then 👍 
But IDE-specific things don't need to be added to a project's gitignore. They can be added to a global one: https://gist.github.com/subfuzion/db7f57fff2fb6998a16c

**Comment by basepi** [comment]
In my opinion, adding IDE-specific gitignore entries is not a problem, as long as they have comments separating them from language-specific gitignore entries. This is not a file where churn (or extra lines) really cost anything.

File: .gitignore [hunk 1]
```diff
@@ -1,3 +1,5 @@
 *.swp
```
Comment (SylvainJuge) [inline .gitignore:1]:
```suggestion
# vim
*.swp
```

File: .gitignore [hunk 2]
```diff
@@ -1,3 +1,5 @@
...
+# IntelliJ module files (.idea folder, *.iml module files):
 .idea
+*.iml
 tools/bin
```
Comment (SylvainJuge) [inline .gitignore:5]:
```suggestion
# project build
tools/bin
```

**Review by SylvainJuge — APPROVED** [review]

**Review by axw — APPROVED** [review]
Agreed, let's merge this with Sylvain's suggestions and make it easy for folks to contribute.

**Pull request merged**